File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.cpp
class Solution {
public:
    int minOperations(string s) {
        
        int n =s.size();
        string p1 ="";
        string p2 ="";
        for(int i=0; i<n;i++){
            if(i%2==0){
                p1+='0';
                p2+='1';
            }
            else{
                p1 +='1';
                p2+='0';

            }
        }
        int change1 =0;
        int change2 =0;
        for(int i=0;i<n;i++){
            if(s[i]!=p1[i]){
                change1++;
            }
            if(s[i]!=p2[i]){
                change2++;
            }
        }
        return min(change1,change2);
    }
};

File: 1758-minimum-changes-to-make-alternating-binary-string/1758-minimum-changes-to-make-alternating-binary-string.py
class Solution:
    def minOperations(self, s: str) -> int:
        n=len(s)
        start0 =0
        start1 =0
        for i in range(n):
            if i%2==0:
                ex0='0'
                ex1='1'
            else:
                ex0 ='1'
                ex1 ='0'
            if s[i]!= ex0:
                start0 +=1
            if s[i]!=ex1:
                start1 +=1
        return min(start0,start1)
        
        